<!DOCTYPE html>
<html>
<head><title>Tooltip Ajax</title></head>
<body>

    <div id="div1">
         <h4>“Quality Control” </h4>
         <p>
        	&nbsp;&nbsp;In “Quality Control” section, every setting is made to control the overall result’s reliability and resolution. The term ‘ambiguous’ refers to n-base, an unknown/unidentified base, hence it is of our best interest to keep the value of ‘maximum ambiguous’ as low as possible. The same trend was applied to the ‘maximum homopolymer’ since homopolymer refers to the repeated bases ( <i class="fa fa-angle-left"></i> 8 is recommended). The minimum and maximum read length recommended for this software is between 100 – 250 base pairs. Although, depending on different dataset, the parameter setting of the software may varies.

        	</p> 

        
    </div>

    <div id="div2">
        <h4>“Align Sequences & Clean Alignment” </h4>
        <p>
        	&nbsp;&nbsp;There are 3 options for the “Alignment Step” in total which are SILVA, Greengenes and RDP. The default was set to be SILVA due to the fact that this database has a better performance at doing alignment than other databases. (REFERENCE) 


        </p>
    </div>

    <div id="div3">
        <h4>“Pre-Cluster Sequences & Chimera Detection” </h4>
        <p>
        	&nbsp;&nbsp;This step is performed to check for the strand similarities, in other words, to check for a possible error in some strands. The main objective of this step is to de-noise the sequences, thus providing a better resolution. To check for errors the recommended number of different bases between strand are ( <i class="fa fa-angle-left"></i> 2. [Pre-Cluster Sequences]
        </p>
             &nbsp;&nbsp;In addition to that, it is possible that the (PCR-based) amplification steps in NGS-platform may cause some errors. This type of errors may cause certain mismatched cases between primer/dimer and their template strand. [Chimera Detection]


        
    </div>

    <div id="div4">
        <h4>“Classify Sequences” </h4>
        <p>
        	&nbsp;&nbsp;To prepare for a taxonomy classification, the a recommended database is Greengenes (set as default). Similar to the reason being the AM software, this consists of a software known as PICRUst, which was designed to predict the function of samples submitted. Furthermore, Greengenes database works at the best performance together with this software. Optionally, if the users are not interested in predicting functions and would like to change the database, SILVA and RDP are also available. 
        </p>
             &nbsp;&nbsp;Greengenes database is the only database that provide information down to species level, while SILVA and RDP can provide only information at the level of genus. In contrast to Greengenes, RDP database was cleaner and SILVA was mostly use for alignment (SILVA > RDP, in terms of taxa). 


        
    </div>

    <div id="div5">
        <h4>“Removal Bacterial Sequences” </h4>
        <p>
        	&nbsp;&nbsp;“Taxon Elimination” is designed to remove the unwanted data (i.e., helping to clean the result). The default was set to eliminate the taxon of chloroplast, mitochondria, eukaryote, and unknown. Optionally, users can eliminate other information, which include an option to remove archaea-unknown and archaea-unknown & bacteria-unknown
        </p>
             
        
    </div>

     <div id="div6">
        <h4>“Alpha – Diversity” </h4>
        <p>
            &nbsp;&nbsp;“Alpha – Diversity” is set as a default setting. This analysis will calculate the statistic for the community of bacteria within the group. A total of 5 statistical analysis will be provided in a table such as nseq, cover, sobs, Chao and Shannon.  
        </p>
             
        
    </div>

    <div id="div7">
        <h4>“Beta – Diversity” </h4>
        <p>
            &nbsp;&nbsp;“Beta – Diversity” will analyze the community of bacteria between different groups. During this step choices to view diagrams for different statistical calculator can be made. A total 7 statistical analysis will be provided in a table, i.e. lennon, jclass, moristahorn, sorabund, thetan, thetayc and braycurtis. 
        </p>
             
        
    </div>

    <div id="div8">
        <h4>“Taxonomy Level” </h4>
        <p>
            &nbsp;&nbsp; In case of phylotype analysis, the default of database for taxonomy classification is Greengenes in genus level. You can change the level of taxonomy in others such as phylum, class, order or species. However, other databases such as RDP and Silva, the species level is not provided. 
        </p> 
    </div>

     <div id="div9">
        <h4>“Amova” </h4>
        <p>
            &nbsp;&nbsp; Analysis of molecular variance (AMOVA) is a non-parametric which was used to test whether the variation between two samples was different from considering the samples as a single population. 
        </p> 
    </div>

     <div id="div10">
        <h4>“Homova” </h4>
        <p>
            &nbsp;&nbsp; Homogeneity of molecular variance (HOMOVA) is a nonparametric analog of Bartlett’s test for homogeneity of variance, which has been used in population genetics to test the hypothesis that the genetic diversity within two or more populations is homogeneous. 
        </p> 
    </div>

    <div id="div11">
        <h4>“Create file design” </h4>
        <p>
            &nbsp;&nbsp; The design file is used to assign samples to a specific treatments, or other categories. 
        </p> 
    </div>


    <div id="div12">
        <h4>“Optional” </h4>
        <p>
            &nbsp;&nbsp;An additional “Option” is provided to help analysing the interaction of bacteria groups within our own samples or with other metadata (abiotic factors such as temperature, salinity and pH). This interaction analysis can be viewed in 2 or 3-dimentions depending on the user. 
        </p>
             
        
    </div>

     <div id="div13">
        <h4>“Create file metadata” </h4>
        <p>
            &nbsp;&nbsp; Metadata file is used to assign samples to other data supports such as experiment factor, sample collection date, pH, temperature, etc.
        </p> 
    </div>

    <div id="div14">
        <h4>“Correlation file metadata ” </h4>
        <p>
            &nbsp;&nbsp; corr.axes was used to correlate metadata such as pH, Temperature, etc. of the subjects in these samples to axes of PCoA or NMDS using the spearman/ pearson’s correlation coefficient.
        </p> 
    </div>

   <div id="div15">
        <h4>“Correlation of each OTU” </h4>
        <p>
            &nbsp;&nbsp; corr.axes was used to correlate bacterial OTUs to axes of PCoA or NMDS using the spearman/pearson’s correlation coefficient.
        </p> 
    </div>

   <div id="div16">
        <h4>“ V1–V3 region with primers 27F and 534R ” </h4>
        <p>
            &nbsp;&nbsp; These primers include hypervariable region V1, V2 and V3 from Zheng et al., 2015 https://link.springer.com/article/10.1186/s40168-015-0110-9.
            <br/>&nbsp;&nbsp;Primers consisted of 27F: 5’ AGAGTTTGATCMTGGCTCAG 3’ and 534R: 5’ ATTACCGCGGCTGCTGG 3’
        </p> 
    </div>

    <div id="div17">
        <h4>“ V3 – V4 region with primers 341F and 802R” </h4>
        <p>
            &nbsp;&nbsp; These primers include hypervariable region V3 and V4. The protocol from illumina platform (https://support.illumina.com/content/dam/illumina-marketing/documents/products/other/16s-metagenomics-faq-1270-2014-003.pdf).
            <br/>&nbsp;&nbsp; Primers consisted of 341F: 5’ CCTACGGGNGGCWGCAG 3’ and 802R: 5’ TACHVGGGTATCTAATCC 3’
        </p> 
    </div>

     <div id="div18">
        <h4>“ V4 region with primers 515F and 806R ” </h4>
        <p>
            &nbsp;&nbsp; These primers with appended 5′ Illumina adapter and 3′ Golay barcode sequences follow the protocol of Caporaso et al. (2012). Detail on http://press.igsb.anl.gov/earthmicrobiome/protocols-and-standards/16s/
            <br/>&nbsp;&nbsp;Primers consisted of 515F : 5′-GTGCCAGCMGCCGCGGTAA-3′ and 806R : 5′-GGACTACHVGGGTWTCTAAT-3′
        </p> 
    </div>

     <div id="div19">
        <h4>“ V3-V5 region with primer 341F and 909R ” </h4>
        <p>
            &nbsp;&nbsp; These primers include hypervariable region in V3, V4 and V5 of bacterial 16S rRNA genes. Example of using these primers in Metzler-Zebeli et al., 2015 http://aem.asm.org/content/81/24/8489.full.pdf
            <br/>&nbsp;&nbsp;Primers consist of 341F 5’ CCTACGGGRSGCAGCAG 3’ and 909R 5’ TTTCAGYCTTGCGRCCGTAC 3’
        </p> 
    </div>

     <div id="div20">
        <h4>“ V4-V5 region with primers 518F and 926R ” </h4>
        <p>
            &nbsp;&nbsp; The detail and description of primers development are published in Nelson et al. (2014). Example of using these primers in Brazelton et al., 2017. https://peerj.com/articles/2945/
            <br/>&nbsp;&nbsp;Primers consisted of 518F : 5’ CCAGCAGCYGCGGTAAN 3’ and 926R : 5’  CCGTCAATTCNTTTRAGT 3’ or 5’ CCGTCAATTTCTTTGAGT 3’ or 5’ CCGTCTATTCCTTTGANT 3’
        </p> 
    </div>





</body>
</html>